pkg/shipper/shipper_e2e_test.go

```diff
@@ -351,3 +351,154 @@ func TestShipper_SyncBlocksWithMigrating_e2e(t *testing.T) {
 		testutil.Assert(t, ok == false, "fifth block was reuploaded")
 	})
 }
+
+// TestShipper_SyncOverlapBlocks_e2e is a unit test for the functionality by allowOutOfOrderUploads flag. This allows compacted(compaction level greater than 1) blocks to be uploaded despite overlapping time ranges.
+func TestShipper_SyncOverlapBlocks_e2e(t *testing.T) {
+	e2eutil.ForeachPrometheus(t, func(t testing.TB, p *e2eutil.Prometheus) {
```

Comment:
Do we need this to be done for each Prometheus version? Sounds quite unnecessary 🤔 I might vote for some cheaper test instead, it all cost in waiting time =D

Reply:
I get your point @bwplotka . Addressed it in a recent commit. 
I added a separate unit test since I thought the other tests in the shipper E2E tests were for similar features. 
To keep the test small, I've only added one case with two blocks.

Reply:
Another reason I added the test in `pkg/shipper/shipper_e2e_tests` is since I couldn't find the shipper tests to add it to in `test/e2ethanos`, hence decided to add it to the existing e2e shipper test.

Reply:
If you feel having the test is extraneous, I can reduce the PR to just the flag condition check.